core/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -165,15 +173,18 @@ public void dispatchRequest(RestRequest request, RestChannel channel, ThreadCont
         }
         RestChannel responseChannel = channel;
         try {
-            int contentLength = request.content().length();
-            if (canTripCircuitBreaker(request)) {
-                inFlightRequestsBreaker(circuitBreakerService).addEstimateBytesAndMaybeBreak(contentLength, "<http_request>");
-            } else {
-                inFlightRequestsBreaker(circuitBreakerService).addWithoutBreaking(contentLength);
+            final int contentLength = request.hasContent() ? request.content().length() : 0;
+            final RestHandler handler = getHandler(request);
+            if (checkForContentType(request, responseChannel, contentLength, handler)) {
```

Comment:
I changed both check methods to simply be boolean responses and the dispatch method is responsible for the sending the response.